source/big_core/illegal_instructions.vh

```diff
@@ -33,16 +33,18 @@ assign PreFunct7Q101H     = PreInstructionQ101H[31:25];
-assign illegal_instructionsQ101H.RopFunct7NotMatchZero = (PreOpcodeQ101H == R_OP) && 
-                                                                                (  PreRtypeQ101H  == SLL_ 
-                                                                                || PreRtypeQ101H  == SLT_  
-                                                                                || PreRtypeQ101H  == SLTU_   
-                                                                                || PreRtypeQ101H  == XOR_  
-                                                                                || PreRtypeQ101H  == OR_   
-                                                                                || PreRtypeQ101H  == AND_) && (PreFunct7Q101H != '0);       
-                                                                                                                 
+// PreFunct7Q101H equals 0x1 in M extension thus not causing illegal instruction
+assign illegal_instructionsQ101H.RopFunct7NotMatchZero = (PreOpcodeQ101H == R_OP && PreFunct7Q101H == 20'h1)    ? 1'b0 : (PreOpcodeQ101H == R_OP) ?
+                                                                                                                         (  PreRtypeQ101H  == SLL_ 
+                                                                                                                         || PreRtypeQ101H  == SLT_  
+                                                                                                                         || PreRtypeQ101H  == SLTU_   
+                                                                                                                         || PreRtypeQ101H  == XOR_  
+                                                                                                                         || PreRtypeQ101H  == OR_   
+                                                                                                                         || PreRtypeQ101H  == AND_) && (PreFunct7Q101H != '0)  : 1'b0;      
+                                                                                                                                                        
 // Covers R_type instructions where bits [31:25] must always be 7'b0100000 = 0x20 or zero in Base ISA //TODO - some extentions may have differente options 
-assign illegal_instructionsQ101H.RopFunct7NotMatch20OrZero = (PreOpcodeQ101H == R_OP) && (PreRtypeQ101H == ADD_ || PreRtypeQ101H  == SRL_) && !(PreFunct7Q101H == 8'h20 || PreFunct7Q101H == '0);
+assign illegal_instructionsQ101H.RopFunct7NotMatch20OrZero = (PreOpcodeQ101H == R_OP && PreFunct7Q101H == 20'h1) ? 1'b0 : (PreOpcodeQ101H == R_OP) ?
+                                                                            (PreRtypeQ101H == ADD_ || PreRtypeQ101H  == SRL_) && !(PreFunct7Q101H == 8'h20 || PreFunct7Q101H == '0) :1'b0;
```

Comment:
I think this needs refactor to be more readable..

I don't understand how the M extension affected this

Reply:
- אכן הכתיבה אינה יפה. הרגשתי ככה אבל לא ידעתי איך לשנות. אשמח שתגיד לי מה לשנות.
- שורה 37 בודקת מצב בו יש אופקוד של R_type אבל funct7 לא היה מתאים כפי שמופיע ב rv32I. פקודות של rv32m הם בעלי אותו opcode רק funt7 שונה לכן הייתי חייב לעדכן את זה כשי שלא יחשוב שמדובר על פקודות מתוך rv32i והם שגויים בגלל ש funt7 לא מתאים.